### PR TITLE
Add tests for #3015

### PR DESCRIPTION
**Pull request by angrykoala**
# Description

Add a test to validate the solution provided for #3015

This issue was not a bug so no fix required for this

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 31 | 49 | 1134 |
|  aggregations.NestedAggregation | 14551 | 14551 | 85 | 109 | 2174 |
|  aggregations.AggregationWithWhere | 11979 | 11979 | 69 | 104 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 22116987 | 2874 | 3079 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116987 | 22116987 | 2780 | 2805 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9734 | 9734 | 77 | 137 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9937 | 9937 | 67 | 74 | 2174 |
|  aggregations.DeeplyNestedCount | 13070329 | 13070349 | 4322 | 4439 | 2008534 |
|  batch-create.BatchCreate | 4200 | 4200 | 157 | 160 | 600 |
|  connect.createAndConnect | 14424 | 14424 | 267 | 195 | 3003 |
|  connections.Connection | 14082 | 14082 | 62 | 93 | 2174 |
|  connections.NestedConnection | 41462 | 41462 | 104 | 113 | 4516 |
|  connections.ConnectionWithSort | 2282 | 2282 | 56 | 67 | 1040 |
|  connections.ConnectionWithSortAndCypher | 2282 | 2282 | 156 | 258 | 1040 |
|  create.SimpleMutation | 7 | 7 | 51 | 60 | 1 |
|  cypher-directive.TopLevelCypherDirective | 116092 | 116092 | 814 | 816 | 12241 |
|  cypher-directive.TopLevelCypherDirectiveWithColumnName | 147609 | 147609 | 269 | 342 | 12241 |
|  delete.SimpleDelete | 19401 | 19401 | 295 | 249 | 1040 |
|  delete.NestedDeleteInUpdate | 22200 | 22200 | 143 | 300 | 2040 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 54 | 77 | 1040 |
|  query.SimpleQuery | 15160 | 15160 | 74 | 88 | 2174 |
|  query.QueryWhere | 9708 | 9703 | 31 | 44 | 2166 |
|  query.SimpleQueryWithNestedWhere | 9886 | 9881 | 53 | 60 | 2166 |
|  query.Nested | 10092037 | 10092037 | 9724 | 10086 | 2008534 |
|  query.NestedWithFilter | 40401 | 40401 | 115 | 158 | 4511 |
|  query.OrFilterOnRelationships | 42553 | 42755 | 151 | 206 | 1607 |
|  query.OrFilterOnRelationshipsAndNested | 35569 | 36293 | 175 | 241 | 1607 |
|  query.QueryWithNestedIn | 13869 | 13456 | 53 | 58 | 1375 |
|  query.NestedConnectionWhere | 9834 | 9834 | 53 | 58 | 2174 |
|  query.DeeplyNestedConnectionWhere | 10054 | 9921 | 67 | 94 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 6343 | 6343 | 121 | 136 | 1134 |
|  query.NestedWithRelationshipSingleFilters | 3024400 | 3024400 | 687 | 559 | 1003942 |
|  query.Fulltext | 96 | 96 | 26 | 28 | 16 |
|  query.FulltextWithNestedQuery | 587 | 587 | 39 | 43 | 84 |
|  sorting.SortMultipleTypes | 2513 | 2513 | 65 | 65 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 1489 | 1487 | 147 | 208 | 1040 |
|  sorting.SortOnNestedFields | 14082 | 14082 | 65 | 85 | 2174 |
|  sorting.SortDeeplyNestedFields | 42196 | 42196 | 114 | 152 | 4516 |
|  sorting.SortWithTopLevelCypher | 2119 | 2119 | 101 | 148 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 67 | 62 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 61 | 66 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 301 | 341 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 401 | 329 | 33033 |
|  update.NestedUpdate | 2119 | 2119 | 62 | 62 | 1040 |

</details>
        
Old Schema Generation: 29.520s
Schema Generation: 29.630s
Old Subgraph Schema Generation: 42.577s
Subgraph Schema Generation: 41.403s

File: packages/graphql/tests/integration/issues/3015.int.test.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+    afterEach(async () => {
+        await session.close();
+    });
+
+    afterAll(async () => {
+        await driver.close();
+    });
+
+    test("custom resolvers should correctly format dates", async () => {
```
Comment (MacondoExpress) [inline packages/graphql/tests/integration/issues/3015.int.test.ts:56]: The test name is wrong.

**Review by MacondoExpress — CHANGES_REQUESTED** [review]
LGTM, I left just a minor comment.

File: packages/graphql/tests/integration/issues/3015.int.test.ts (hunk 1 above)
Comment (angrykoala) [reply packages/graphql/tests/integration/issues/3015.int.test.ts:56]: Good catch

```suggestion
    test("union should filter by top level match", async () => {
```

**Review by MacondoExpress — APPROVED** [review]

**Pull request merged**